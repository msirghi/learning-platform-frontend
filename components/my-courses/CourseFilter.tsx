import React from 'react';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';
import FilterListIcon from '@material-ui/icons/FilterList';
import styles from '../../styles/modules/MyCourses.module.scss';
import { useTranslation } from '../../i18n';

type Props = {
  onSearchInputChange: (val: string) => void;
  searchValue: string;
};

export const CourseFilter: React.FC<Props> = ({ searchValue, onSearchInputChange }) => {
  const { t } = useTranslation();
  return (
    <div className={styles.myCoursesHeaderRightWrapper}>
      <TextField
        size='small'
        label={t('courses:searchCourse')}
        onChange={(e) => onSearchInputChange(e.target.value)}
        value={searchValue}
      />
      <IconButton>
        <FilterListIcon />
      </IconButton>
    </div>
  );
};
