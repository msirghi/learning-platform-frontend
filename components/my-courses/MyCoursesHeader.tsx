import React from 'react';
import styles from '../../styles/modules/MyCourses.module.scss';
import BulbIcon from '@material-ui/icons/EmojiObjects';
import { CourseFilter } from './CourseFilter';

type Props = {
  onSearchInputChange: (val: string) => void;
  searchValue: string;
};

export const MyCoursesHeader: React.FC<Props> = ({ onSearchInputChange, searchValue }) => {
  return (
    <div className={styles.myCoursesHeader}>
      <div className={styles.myCoursesHeaderLeftWrapper}>
        <BulbIcon />
        <span>Lorem ipsum dolor sit amet.</span>
      </div>
      <CourseFilter onSearchInputChange={onSearchInputChange} searchValue={searchValue} />
    </div>
  );
};
