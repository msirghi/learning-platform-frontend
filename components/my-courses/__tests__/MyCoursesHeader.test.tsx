import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { CourseFilter } from '../CourseFilter';
import { MyCoursesHeader } from '../MyCoursesHeader';

describe('MyCoursesHeader component', () => {
  const props = {
    onSearchInputChange: jest.fn(),
    searchValue: 'string'
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<MyCoursesHeader {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onSeachInputChange on filter change', () => {
    const wrapper = shallow(<MyCoursesHeader {...props} />);
    const filter = wrapper.find(CourseFilter);

    expect(filter).toHaveLength(1);
    filter.props().onSearchInputChange('val');
    expect(props.onSearchInputChange).toBeCalled();
    expect(props.onSearchInputChange).toBeCalledWith('val');
  });
});
