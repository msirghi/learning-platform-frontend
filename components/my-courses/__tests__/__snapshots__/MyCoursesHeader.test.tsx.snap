// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCoursesHeader component should match the snapshot 1`] = `
<div
  className="myCoursesHeader"
>
  <div
    className="myCoursesHeaderLeftWrapper"
  >
    <Memo([object Object]) />
    <span>
      Lorem ipsum dolor sit amet.
    </span>
  </div>
  <CourseFilter
    onSearchInputChange={[MockFunction]}
    searchValue="string"
  />
</div>
`;
